source/extensions/filters/http/grpc_http1_bridge/http1_bridge_filter.cc

```diff
@@ -19,6 +19,19 @@ namespace Extensions {
 namespace HttpFilters {
 namespace GrpcHttp1Bridge {
 
+// Some client requests' URLs may contain query params. gRPC upstream servers can not
+// handle these requests, and may return error such as "unknown method". So we remove
+// query params here.
+void Http1BridgeFilter::removeQueryParams(Http::RequestHeaderMap& headers) {
+  absl::string_view path = headers.Path()->value().getStringView();
+  size_t pos = path.find("?");
+  if (pos != absl::string_view::npos) {
+    absl::string_view new_path = path.substr(0, pos);
+    headers.setPath(new_path);
+    decoder_callbacks_->downstreamCallbacks()->clearRouteCache();
```

Comment:
Do you really want to do this? I'm not sure if this would be expected when "ignoring" parameters.

Reply:
> Yeah.. you remind me. Re-routing is not necessary. I will fix this.

Fixed already.